ast.go

```diff
@@ -652,6 +657,37 @@ func (fa *ASTFunc) addCToGoSliceConversion(name string, cname string, lengthOfSl
 	})
 }
 
+func (fa *ASTFunc) addCArrayFromGoSlice(name string, cname string, typ Type) {
+	sliceType := getSliceType(typ)
+
+	fa.addAssignment("gos_"+name, &ast.CallExpr{
```

Comment:
Should we maybe do this in just one generated line?

```go
cp_unsavedFiles := (*C.struct_CXUnsavedFile)(unsafe.Pointer((*reflect.SliceHeader)(unsafe.Pointer(&unsavedFiles)).Data))
```

The "gos_" variable is used used only once temporarily.